lib/rules/selector-max-universal/index.js

```diff
@@ -46,7 +59,21 @@ const rule = (primary) => {
 					checkSelector(childNode, ruleNode);
 				}
 
-				if (childNode.type === 'universal') total += 1;
+				let prevChildNode = childNode.prev();
+				let prevChildNodeValue = prevChildNode && prevChildNode.value;
+
+				// checks if previous child node is descendant combinator
+				if (prevChildNode && prevChildNode.type === 'combinator' && prevChildNodeValue === ' ') {
+					prevChildNode = prevChildNode.prev();
+
+					prevChildNodeValue = prevChildNode && prevChildNode.value;
+				}
```

Comment:
What purpose does this bit of code have?

Reply:
> What purpose does this bit of code have?

As you know, there are 4 types of combinators, according to [MDN](https://developer.mozilla.org/en-US/docs/Learn/CSS/Building_blocks/Selectors/Combinators). This code's purpose is to add support for descendant combinator. This way user can add smth like this to the configuration: `["h1"]`.

For example, we have code like this: `a > * { color: red }`. With combinators `">", "+", "~"` it's very easy to check, in this case we did it this way: 
`let prevChildNode = childNode.prev();`
`let prevChildNodeValue = prevChildNode && prevChildNode.value;`
`prevChildNodeValue` will equal to `">"`, `"+"` or `"~"`, and then we just pass the value to optionMatches function:
`optionsMatches(secondaryOptions, 'ignoreAfterCombinators', prevChildNodeValue)`

If we have code `h1 * { color: red }` and we want to ignore `*` after `h1`, we will stumble upon descendant combinator, which is represented by a single space (" ") character. Inn this case, the code from above will not work as `prevChildNoveValue` will equal to `" "`. We have to go back even farther and check previous node of the prevChildNode to get the value `h1`.

Reply:
Thanks for the explanation. I think we can remove this code.

> This code's purpose is to add support for the descendant combinator.

I believe the rule supports the descendant combinator without this code, via the configuration of `[" "]`. We could probably do more to accommodate other whitespace descendant combinators (like tabs and newlines), but let's only do that if someone requests it. By supporting `">"`, `"+"`, `"~"` and `" "` (space) we satisfy the original use case in https://github.com/stylelint/stylelint/issues/5792.

> If we have code `h1 * { color: red }` and we want to ignore `*` after `h1`

We can introduce a separate `ignoreAfterSelector: []` secondary option to support this as `h1` is a [type selector](https://drafts.csswg.org/selectors/#type-selectors), not a combinator (and `["h1"]` shouldn't be a valid configuration for the `ignoreAfterCombinators` option). Let's only add this option if someone requests it, though.